test/unit/vl-map-draw-action.test.html

```diff
@@ -0,0 +1,60 @@
+<!doctype html>
+<html>
+
+<head>
+  <meta charset="utf-8">
+  <script src="../../../@webcomponents/webcomponentsjs/webcomponents-lite.js"></script>
+  <script src="../../../web-component-tester/browser.js"></script>
+
+  <script type="module" src="../../src/vl-map-all.js"></script>
+  <script src="/node_modules/sinon/pkg/sinon.js"></script>
+</head>
+
+<test-fixture id="vl-map-draw-actions-fixture">
+  <template>
+    <vl-map>
+      <vl-map>
+        <vl-map-features-layer id="line-layer">
+          <vl-map-draw-point-action id="draw-point-action"></vl-map-draw-point-action>
+          <vl-map-draw-line-action id="draw-line-action"></vl-map-draw-line-action>
+          <vl-map-draw-polygon-action id="draw-polygon-action"></vl-map-draw-polygon-action>
+        </vl-map-features-layer>
+      </vl-map>
+    </vl-map>
+  </template>
+</test-fixture>
+<body>
+  <script type="module">
+    import {awaitUntil} from '/node_modules/vl-ui-core/dist/vl-core.js';
+
+    suite('vl-map-draw-line-action', () => {
+      setup(async () => {
+        await customElements.whenDefined('vl-map-draw-point-action');
+        await customElements.whenDefined('vl-map-draw-line-action');
+        await customElements.whenDefined('vl-map-draw-polygon-action');
+      });
+
+      test('nadat de tekenactie voltooid is, zal de callback worden opgeroepen', async () => {
+        const map = fixture('vl-map-draw-actions-fixture');
+        await map.ready;
+        const types = ['point', 'line', 'polygon'];
+        for (let index = 0; index < types.length; index++) {
+          const drawAction = map.querySelector(`vl-map-draw-${types[index]}-action`);
+          await awaitUntil(() => drawAction.action != null);
+          let featureDrawn = false;
+          drawAction.onDraw((feature) => {
+            assert.equal(feature, 'getekendeFeature');
+            featureDrawn = true;
+          });
+          drawAction.action.drawInteraction.dispatchEvent({
+            type: 'drawend',
+            feature: 'getekendeFeature',
+          });
+          await awaitUntil(() => featureDrawn === true);
```

Comment:
Niet beter hier de done functie van uw test te gebruiken en die pas aan te roepen in de onDraw?

Reply:
De `done` kan niet gecombineerd worden met `await`, dan moeten de awaits vervangen worden door then implementatie, wat natuurlijk ook kan. Lijkt me puur stijl voorkeur, dus wat mij betreft zijn beide opties prima.

Reply:
Het is op zich nog niet done hier, want er is een for-loop met de drie teken acties.